### PR TITLE
(PUP-7583) Remove to_yaml_properties

### PR DESCRIPTION
**Pull request by thallgren**
This commit removes the `#to_yaml_properties` method everywhere in the
Puppet code-base and adds an include of `Puppet::Util::PsychSupport`
where needed.

Some tests that only asserted the returned value from the removed
method was also removed. Other tests were rewritten so that they use
the keys of the hash returned from the `#to_data_hash` methods instead.

### CODE REVIEW

File: lib/puppet/graph/simple_graph.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 # A hopefully-faster graph class to replace the use of GRATR.
 class Puppet::Graph::SimpleGraph
+  include Puppet::Util::PsychSupport
```
Comment (joshcooper) [inline lib/puppet/graph/simple_graph.rb]: Can you break this out into a separate PR for PUP-7381, since that one is about using `encode_with` consistently, and this PR is about removing duplication?

Comment (thallgren) [inline lib/puppet/graph/simple_graph.rb]: Done. Please note that this PR is on top of the PR for PUP-7381.

Comment (thallgren) [inline lib/puppet/graph/simple_graph.rb]: I verified that the puppetdb unit tests are green.
